library/psa_crypto_slot_management.c

```diff
@@ -226,37 +226,76 @@ psa_status_t psa_validate_key_persistence( psa_key_lifetime_t lifetime,
     }
 }
 
+#if defined(MBEDTLS_PSA_CRYPTO_BUILTIN_KEYS)
+static psa_status_t get_builtin_key( mbedtls_svc_key_id_t svc_key_id,
+                                     psa_key_slot_t *slot )
+{
+    psa_key_id_t app_key_id = MBEDTLS_SVC_KEY_ID_GET_KEY_ID( svc_key_id );
+    if( ! ( app_key_id >= MBEDTLS_PSA_KEY_ID_BUILTIN_MIN &&
+            app_key_id <= MBEDTLS_PSA_KEY_ID_BUILTIN_MAX ) )
+    {
+        return( PSA_ERROR_DOES_NOT_EXIST );
+    }
+
+    psa_key_attributes_t attributes = PSA_KEY_ATTRIBUTES_INIT;
+    psa_set_key_id( &attributes, svc_key_id );
+    psa_set_key_lifetime( &attributes, PSA_KEY_LIFETIME_VOLATILE );
```

Comment:
This may be a volatile copy of the key material, or a volatile object that contains a reference to the key material. Either way, the key object itself is volatile. The key material is permanent since you can make a new object to access the same key material at any time.